elytron/ELY-201-Add-client-side-HTTP-authentication-mechanism-support.adoc

```diff
@@ -0,0 +1,171 @@
+
+[source,xml]
+----
+<set-user-name name="quickstartUser"/>
+<credentials>
+   <clear-password password="quickstartPwd1!"/>
+</credentials>
+----
+
+We will use different type of authentication mechanism 
+Basic,bearer,certificate,digest,external
+
+Based on the mechanism type we will call different methods to get authorization value which will be added to the header for authentication.
+Like for basic type authentication mechanism we will create a class which contains evaluateMechanism() method :  
+
+[source,xml]
+----
+public static HttpRequest evaluateMechanism(URI uri) {
+    HttpRequest request = HttpRequest
+            .newBuilder()
+            .uri(uri)
+            .header(ElytronHttpClientConstants.AUTHORIZATION, basicAuth(userName, password))
+            .build();
+    return request;
+}
+----
+
+Similar to this we will have different classes for different authentication mechanism like digest,bearer,etc.
+
+Now after sending the request we will get a response which is of type HttpResponse.we will determine using the status if the authentication success or failed.If status code is 200 then we will know that authentication is successful.
+
+If the authentication failure then headers will contain : 
+www-authenticate=[Basic realm="RealmUsersRoles"] for basic authentication.similar for the other authentication mechanism type like digest,bearer,etc.
+
+
+== Security Considerations
+
+////
+Identification if any security implications that may need to be considered with this feature
+or a confirmation that there are no security implications to consider.
+////
+
+== Test Plan
+
+To test the above functionality we can add the tests as follows : 
+Inside tests/base/src/test/java/org/wildfly/security/http/client/ we will create a test class named as ElytronHttpClientTest where we will get an instance of HttpRequest from ElytronHttpClient#evaluateMechanism() method of different authentication mechanism and then we will create an object of TestingHttpServerRequest using request header which we will pass to the evaluateRequest() method of BasicAuthenticationMechanism which is used to evaluate our header values of request method.
+
+[source,xml]
+----
+TestingHttpServerRequest testingHttpServerRequest = new TestingHttpServerRequest(new String[]{request.headers().allValues("Authorization").get(0)});
+----
+
+The actual communication flow and connect method have be tested with arquillian in wildfly where we will call the connect method with required parameter and test it.
+
+== Community Documentation
+////
+Generally a feature should have documentation as part of the PR to wildfly master, or as a follow up PR if the feature is in wildfly-core. In some cases though the documentation belongs more in a component, or does not need any documentation. Indicate which of these will happen.
+////
+== Release Note Content
+////
+Draft verbiage for up to a few sentences on the feature for inclusion in the
```

Comment:
Try to draft here a release note content. That should include a brief description of this feature